src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -202,234 +206,173 @@ const FixPhoneNumbers: React.FC<Props> = ({
-        });
-      },
     });
   };
 
-  const handleDelete = (
-    values: FormValues,
-    setValues: (values: FormValues) => void,
-  ): void => {
-    const temp = JSON.parse(JSON.stringify(values));
-
-    const deleting = temp?.people[
-      deleteModalState.personIndex
-    ].phoneNumbers?.nodes.splice(deleteModalState.numberIndex, 1)[0];
-
-    deleting.destroy = true;
-
-    if (
-      deleting.primary &&
-      temp?.people[deleteModalState.personIndex]?.phoneNumbers?.nodes.length
-    ) {
-      temp.people[deleteModalState.personIndex].phoneNumbers.nodes[0].primary =
-        true;
+  const handleChangePrimary = (personId: string, numberIndex: number): void => {
+    const temp = { ...dataState };
+    if (temp[personId]) {
+      temp[personId].phoneNumbers = temp[personId].phoneNumbers.map(
+        (number, index) => ({
+          ...number,
+          primary: index === numberIndex,
+        }),
+      );
     }
-
-    setValues(temp);
-    handleDeleteModalClose();
+    setDataState(temp);
   };
 
-  const fixPhoneNumberSchema = yup.object({
-    people: yup.array().of(
-      yup.object({
-        phoneNumbers: yup.object({
-          nodes: yup.array().of(
-            yup.object({
-              id: yup.string().nullable(),
-              number: yup.string().when('destroy', {
-                is: true,
-                then: yup.string().nullable(),
-                otherwise: yup
-                  .string()
-                  .required(t('This field is required'))
-                  .nullable()
-                  .test(
-                    'is-phone-number',
-                    t('This field is not a valid phone number'),
-                    (val) => typeof val === 'string' && /\d/.test(val),
-                  ),
-              }),
-              destroy: yup.boolean().default(false),
-              primary: yup.boolean().required('please select a primary number'),
-              historic: yup.boolean().default(false),
-            }),
-          ),
-        }),
-        isNewPhoneNumber: yup.boolean().default(false),
-        newPhoneNumber: yup.string().when('isNewPhoneNumber', {
-          is: false,
-          then: yup.string().nullable(),
-          otherwise: yup
-            .string()
-            .required(t('This field is required'))
-            .nullable()
-            .test(
-              'is-phone-number',
-              t('This field is not a valid phone number'),
-              (val) => typeof val === 'string' && /\d/.test(val),
-            ),
-        }),
-      }),
-    ),
-  });
-
   return (
     <Box className={classes.container}>
-      {!loading && data ? (
-        <>
-          {data?.people.nodes.length > 0 ? (
-            <Formik
-              initialValues={initialValues}
-              onSubmit={() => {}}
-              validationSchema={fixPhoneNumberSchema}
-            >
-              {({ errors, setValues, values }): ReactElement => (
+      {data && dataState ? (
```

Comment:
`dataState` can't ever be undefined, right?

```suggestion
      {data ? (
```